md/musig2-adaptorsig.md

```diff
@@ -0,0 +1,230 @@
+# Takes other signer's public key B and other signer's nonces from setupSession()
+# RB1, RB2 and message m and adaptor T.
+preSign_{x,state}((B, RB1, RB2), m, T):
+    # Similar to regular MuSig signing, the same state must never be
+    # reused for different preSign calls
+    (muA, P) := keyAgg(x*G, B)
+    (rA1, RA1, rA2, RA2) := state
+    R1 := RA1 + RB1, R2 := RB1 + RB2
+    b := hash(P, T + R1, R2, m)
+    R := R1 + b*R2
+    rA := rA1 + b*rA2
+    pre_s := rA + hash(P, R + T, m)*muA*x
+    return pre_s
+
+# Verify pre-signature pre_s for public key B, nonces RB1, RB2 and the
+# signer's public key A, the signer's nonces RA1 and RA2, message m
+# and adaptor T.
+preVerify_{B,RB1,RB2}((A, RA1, RA2), m, T, pre_s):
+    (muA, P) := keyAgg(A, B)
+    R1 := RA1 + RB1, R2 := RB1 + RB2
+    b := hash(P, T + R1, R2, m)
+    R := R1 + b*R2
+    RA := RA1 + b*RA2
+    return pre_s*G ?= RA + hash(P, R + T, m)*muA*A
+
+Adapt(pre_s, t):
+    s := pre_s + t
+    return s
+
+Ext(s, pre_s):
+    return s - pre_s
+```
+
+Note that another option would be to compute `b` by concatenating the adaptor `T` to the string that is hashed instead of adding it to the nonces: `b = hash(P, R1, R2, m, T)`
+But this has the downside of revealing to every participant in the multisig that an adaptor signature protocol is being executed
+For example, if there is a MuSig setup with three signers and only two of them are making use of adaptor signatures, then this option would require the third signer to know about the adaptor to compute `b`.
+
+### Correctness
+```
+let
+    # Alice sets up session
+    (state, RA1, RA2) := setupSession()
+    # Bob sets up session
+    (_, RB1, RB2) := setupSession()
+    # Alice may obtain RB1, RB2 before generating A
+    (x, A) := KeyGen()
+    (_, B) := KeyGen()
+    # Alice creates a pre-signature for Bob
+    pre_s := preSign_{x,state}((B, RB1, RB2), m, T)
+    s := Adapt(pre_s, t)
+    t' := Ext(s, pre_s)
+    Verify_{B,RB1,RB2}((A, RA1, RA2), m, s):
+        # MuSig2 partial sig verification. If passing, Bob can complete it to
+        # a Schnorr signature.
+        preVerify_{B,RB1,RB2}((A, RA1, RA2), m, 0, s):
+
+then
+    preVerify_{B,RB1,RB2}((A, RA1, RA2), m, T, pre_s) = true
+    Verify_{B,RB1,RB2}((A,RA1+T,RA2), m, s) = true
+    t'*G = T
+```
+
+## Security
+
+What follows are security definition and proof sketches for adaptor signatures following [Generalized Bitcoin-Compatible Channels](https://eprint.iacr.org/2020/476)
+They are not too different to the proof for Schnorr adaptor signatures in that publication, but does not require strong unforgeability of the underlying signature scheme
+The sketches are intended to test above definition of the scheme, give some intuition about what adaptor signatures tries to achieve and should give a bit more confidence as long as there exists no complete proof.
+
+### aExistential Unforgeability under Chosen Message Attack
```

Comment:
extra `a` here

Reply:
That's actually the name according to the paper. Perhaps `a` stands for adaptor? Should probably  add `[sic]`.